src/coreclr/jit/gentree.cpp

```diff
@@ -7983,22 +7983,24 @@ GenTree* Compiler::gtClone(GenTree* tree, bool complexOK)
             break;
 
         case GT_LCL_VAR:
-            // Remember that the LclVar node has been cloned. The flag will be set
-            // on 'copy' as well.
-            tree->gtFlags |= GTF_VAR_CLONED;
+        case GT_LCL_VAR_ADDR:
```

Comment:
Adding `GT_LCL_VAR_ADDR` is not part of the fix, but I just could not unsee how we clone `LCL_FLD_ADDR` but not its offset-less equivalent.